### PR TITLE
CNF-12400: Enhance seedgen system check with additional requirement checking

### PR DESCRIPTION
**Pull request by leo8a**
The seedgen system check validates the configuration of the cluster being used to generate a seed image to ensure it meets certain prerequisites, such as ensuring the cluster is not currently managed by ACM. 

This PR extends this validation with additional checks to avoid generating seed images that may include problematic configurations that can cause issues for users later on during an IBU.

Additional checks added in system validation prior to seed creation:

- Ensure no PVs are configured
- Ensure no localvolume CRs exist, if lso is used
- Ensure no lvmcluster CR exists, if lvms is used
- Ensure DataProtectionApplication CR does not exist, if OADP is used
- Ensure minimum OADP version (i.e., >=1.3.1 for now), if OADP is used

/cc @donpenney @Missxiaoguo @pixelsoccupied @browsell

### CODE REVIEW

**Comment by leo8a** [comment]
/hold
(testing with different storage backends)

**Comment by leo8a** [comment]
/unhold

PTL when u guys get a chance

**Comment by pixelsoccupied** [comment]
lgtm

**Comment by donpenney** [comment]
The e2e test is failing due to the OADP operator CSV missing:
```
2024-07-03T12:56:36.046039510Z 2024-07-03T12:56:36Z     INFO    controllers.SeedGenerator       Seed generation rejected: system validation failed: Failure occurred when checking OADP version in system validation: failed to find CSV for OADP operator
```

File: internal/backuprestore/common.go [hunk 1]
```diff
@@ -535,6 +522,26 @@ func (h *BRHandler) CheckOadpOperatorAvailability(ctx context.Context) error {
 
 // IsOadpInstalled a simple function to determine if OADP is present by checking the presence of at least one OADP defined CRD
```
Comment (pixelsoccupied) [inline internal/backuprestore/common.go:523]: I think we can update this comment to reflect the improvements

**Pull request merged**